core/src/cgcloud/core/box.py

```diff
@@ -406,8 +552,47 @@ def create( self, wait_ready=True, cluster_ordinal=0 ):
                 result.append( box )
         except StopIteration:
             pass
+
         return result
 
+    def _spot_create( self ):
+        """
+        A request for multiple instances actually creates many requests for 1 instance each,
+        which then become reservations with 1 instance each.
+
+        :rtype: list[boto.ec2.instance.Reservation]
+        """
+        requests = self.ctx.ec2.request_spot_instances( image_id=self.image_id,
+                                                        **self.instance_creation_args )
+        instances = [ ]
+
+        instance_ids = set()
+        try:
+            while True:
+                updatedRequests = self.ctx.ec2.get_all_spot_instance_requests(
```

Comment:
Here you are asking for a status update for every one of the original requests, even the ones you know have been already fulfilled.